src/DFG.cpp

```diff
@@ -175,27 +175,30 @@ void DFG::combine(string t_opt0, string t_opt1) {
   }
 }
 
-void DFG::combineForIter(string* t_targetPattern, int patternSize, list<DFGNode*>* t_matchedNodes){
-  // DFG::combineForIter combines patterns provided by the user, which should be a cycle. Otherwise, the fusion won't be performed.
-  string headOpt = t_targetPattern[0];
-  for (DFGNode* headNode: nodes) {
-    if (headNode->isOpt(headOpt) and !headNode->hasCombined()) {
-      t_matchedNodes->push_back(headNode);
-      for (int i = 1; i < patternSize; i++){
-        string t_opt = t_targetPattern[i];
+void DFG::combineForIter(list<string>* t_targetPattern){
+  int t_patternSize = t_targetPattern->size();
+  string t_headOpt = string(t_targetPattern->front());
+  list<string>::iterator t_patternNow = t_targetPattern->begin();
```

Comment:
Rename it to `currentFunc`? Pattern indicates a set of Func in your context, right?